analysis/priestholy/src/modules/shadowlands/items/FlashConcentration.tsx

```diff
@@ -0,0 +1,228 @@
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL),
+      this.onByPlayerHealAmount,
+    );
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL),
+      this.onByPlayerHealCast,
+    );
+  }
+
+  /**
+   * Sets _isFullStack to true if there are now 5 stacks, as well as the last refresh
+   * NB: it comes after the Flash Heal Cast
+   * @param event indicate the new number of stacks (only positive increments)
+   */
+  onByPlayerFlashConcentrationBuff(event: ApplyBuffStackEvent) {
+    if (event.stack === 5) {
+      this._isFullStack = true;
+      this._lastFullStackRefresh = event.timestamp;
+    } else {
+      this._isFullStack = false;
+    }
+    this._currentStacks = event.stack;
+  }
+
+  /**
+   * Sets _isFullStack to false.
+   * Note: The only exception is the prepull event, for which we assume that
+   * it means the character has five stacks. It will be corrected if the
+   * first Flash Heal generates a ApplyBuffStack or ApplyBuff event
+   * Note 2: Technically, the only time this function does something is on the first
+   * apply buff if there was no prepull FC.
+   * @param event with info about first stack, or prepull incomplete indication
+   */
+  onByPlayerFlashConcentrationInitialBuff(event: ApplyBuffEvent) {
+    if (event.prepull !== true) {
+      this._isFullStack = false;
+      this._currentStacks = 1;
+    }
+  }
+
+  /**
+   * Increments _totalDrops for our suggestion, since that should ideally not happen
+   */
+  onByPlayerFlashConcentrationBuffRemoval(event: RemoveBuffEvent) {
+    this._totalDrops += 1;
+    this._isFullStack = false;
+    this._currentStacks = 0;
+  }
+
+  /**
+   * when at full stack, it refreshes the last refresh timer,
+   * and counts a Wasteful Cast if it was used earlier than needed
+   */
+  onByPlayerFlashHealCast(event: CastEvent) {
+    if (this._isFullStack) {
+      if (event.timestamp - this._lastFullStackRefresh < 15000) {
+        this._wastefulCasts += 1;
+      }
+      this._lastFullStackRefresh = event.timestamp;
+    }
+  }
+
+  onByPlayerHealAmount(event: HealEvent) {
+    this._totalHeal += event.amount;
+    this._gainedHeal +=
+      (event.amount * (this._currentStacks * 0.03)) / (1 + this._currentStacks * 0.03);
+    this._gainedCastTime += this._currentStacks * 200;
+  }
+
+  onByPlayerHealCast(event: CastEvent) {
+    this._totalCastTime +=
+      event.timestamp -
+      (event.channel?.beginChannel?.timestamp
+        ? event.channel.beginChannel.timestamp
+        : event.timestamp);
+  }
+
+  statistic() {
+    // HealHPSNoFC gives the HPS which would have been obtained by casting the same number of heals without Flash Concentration Buff
+    const HealHPSNoFC =
+      (this._totalHeal - this._gainedHeal) / (this._totalCastTime + this._gainedCastTime);
+    // BuffedHealHPS gives the total HPS of Heal in this fight
+    const BuffedHealHPS = this._totalHeal / this._totalCastTime;
```

Comment:
I think this calculation is really sketchy and you shouldn't be trying to mash together the heal gain and cast times like this. Healers are mana limited and so we can't directly attribute cast time like that.

I'm up for more discussion on this, but I think you split the healing gained from the buff and the cast time gains separately. Alternately, you could pretend a fully buffed Heal is like a stronger and cheaper Flash Heal and tally it that way.

Reply:
I am expecting discussion on this one for sure. I put it into the tooltip, but Flash Concentration fundamentally changes the Gameplay of Holy priest, so hard to compare. I thought about your proposal to compare to Flash heal casts, but it sounded weirder in my mind, because that's another spell, and Heal replaces more then only Flash Heals casts under Flash Concentration (see below). 
My first iteration was to have the gain in hps from using the same amount of heal casts, unbuffed, without taking into account the cast time gain. And to be honest it was irrelevant. like 150HPS, which is misleading. 
so what I did hear is compare what the HPS produced by the same amount of Heal Casts would have been without FC, to what we see with FC. And then I give what it means gained HPS over the total time spent casting Heal. 
It is absolutely not perfect, but that is the most relevant calculation I could find. the reality is that using Flash Concentration makes Heal replace:
- Flash Heal
- Heal 
- Prayer of Healing
- Renew ...
So the impact is very hard to value :man_shrugging: 
I am open to suggestion for sure. More theorycrafting experience would be very valuable here as well

Reply:
Generally I think if you can't make a reasonable statistic, you shouldn't. It's possible there's just no satisfying thing to display for this legendary, and that's OK. You could just add in the suggestions and leave it at that, or add a less ambitious statistic instead.